src/transformers/generation/utils.py

```diff
@@ -441,6 +441,9 @@ def _maybe_initialize_input_ids_for_generation(
             if isinstance(value, torch.Tensor):
                 batch_size = value.shape[0]
                 break
+
+        if "inputs_embeds" in model_kwargs:
+            return torch.ones((batch_size, 0), dtype=torch.long, device=self.device)
```

Comment:
For my own understanding - am I correct in understanding when using `input_embeds` we don't use any initialization then, this is just an empty placeholder?

Reply:
Yep, when we initialized with size 1 filled with BOS tokens, that ruined max_length by one token. We want want the final generation be a continuation of `input_embeds` and not start with BOS